CMakeLists.txt

```diff
@@ -35,6 +35,10 @@ project(OpenVINOGenAI
         HOMEPAGE_URL "https://github.com/openvinotoolkit/openvino.genai"
         LANGUAGES CXX C)
 
+set(CMAKE_CXX_STANDARD 17)
+set(CMAKE_CXX_STANDARD_REQUIRED ON)
+set(CMAKE_CXX_EXTENSIONS OFF)        
```

Comment:
why is it required? 
we already have such properties here:
https://github.com/openvinotoolkit/openvino.genai/blob/6eccb0980ba688c8b89c5adc7f8fc67d3ab70d7f/src/cpp/CMakeLists.txt#L105